src/test/java/itest/GraphQLIT.java

```diff
@@ -588,6 +601,426 @@ void testNodesHaveIds() throws Exception {
+        webClient
+                .post(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .sendForm(
+                        form1,
+                        ar -> {
+                            if (assertRequestStatus(ar, createRecordingFuture1)) {
+                                createRecordingFuture1.complete(null);
+                            }
+                        });
+        createRecordingFuture1.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        CompletableFuture<Void> createRecordingFuture2 = new CompletableFuture<>();
+        MultiMap form2 = MultiMap.caseInsensitiveMultiMap();
+        form2.add("recordingName", "Recording2");
+        form2.add("duration", "5");
+        form2.add("events", "template=ALL");
+        webClient
+                .post(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .sendForm(
+                        form2,
+                        ar -> {
+                            if (assertRequestStatus(ar, createRecordingFuture2)) {
+                                createRecordingFuture2.complete(null);
+                            }
+                        });
+        createRecordingFuture2.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        // GraphQL Query to filter Active recordings by names
+        CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
+        String query =
+                "query { targetNodes (filter: {name:"
+                    + " \"service:jmx:rmi:///jndi/rmi://cryostat-itests:9091/jmxrmi\"}){ recordings"
+                    + " {active(filter: { names: [\"Recording1\", \"Recording2\",\"Recording3\"] })"
+                    + " {data {name}}}}}";
+        webClient
+                .post("/api/v2.2/graphql")
+                .sendJson(
+                        new JsonObject().put("query", query),
+                        ar -> {
+                            if (assertRequestStatus(ar, resp2)) {
+                                resp2.complete(
+                                        gson.fromJson(
+                                                ar.result().bodyAsString(),
+                                                TargetNodesQueryResponse.class));
+                            }
+                        });
+
+        TargetNodesQueryResponse graphqlResp = resp2.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        List<ActiveRecording> filteredRecordings = new ArrayList<>();
+        List<String> filterNames = Arrays.asList("Recording1", "Recording2");
+
+        for (TargetNode targetNode : graphqlResp.data.targetNodes) {
+            List<ActiveRecording> recordings = targetNode.recordings.active.data;
+            for (ActiveRecording recording : recordings) {
+                if (filterNames.contains(recording.name)) {
+                    filteredRecordings.add(recording);
+                }
+            }
+        }
```

Comment:
What is this for-loop needed for? It seems like it is providing a filtered version of the query response that only contains the expected data, and then the test continues to assert that this filtered version matches the expectations - but it has to, because this loop is trimming the response down to the expectation first. I think the assertions below can just be done directly on the query response, can't they?

Reply:
I got rid of the for loop and assert directly